cargo-verify/src/main.rs

```diff
@@ -0,0 +1,841 @@
+    stderrlog::new().verbosity(opt.verbosity).init()?;
+
+    if opt.clean {
+        info_at!(&opt, 1, "Running `cargo clean`");
+        Command::new("cargo")
+            .arg("clean")
+            .arg("--manifest-path")
+            .arg(&opt.cargo_toml)
+            .output()
+            .ok(); // Discarding the error on purpose.
+    }
+
+    let package = get_meta_package_name(&opt)?;
+    info_at!(&opt, 1, "Checking {}", &package);
+
+    let status = match opt.backend {
+        Backend::Proptest => {
+            info_at!(&opt, 1, "  Invoking cargo run with proptest backend");
+            proptest::run(&opt)
+        }
+        _ => {
+            let target = get_default_host(&opt.cargo_toml.parent().unwrap_or(&PathBuf::from(".")))?;
+            info_at!(&opt, 4, "target: {}", target);
+            verify(&opt, &package, &target)
+        }
+    }
+    .unwrap_or_else(|err| {
+        error!("{}", err);
+        exit(1)
+    });
+
+    println!("VERIFICATION_RESULT: {}", status);
+    if status != Status::Verified {
+        exit(1);
+    }
+    Ok(())
+}
+
+// Compile a Rust crate to generate bitcode
+// and run one of the LLVM verifier backends on the result.
+fn verify(opt: &Opt, package: &str, target: &str) -> CVResult<Status> {
+    // Compile and link the patched file using LTO to generate the entire
+    // application in a single LLVM file
+    info_at!(&opt, 1, "  Building {} for verificatuin", package);
+    let bcfile = build(&opt, &package, &target)?;
+
+    // Get the functions we need to verify, and their mangled names.
+    let tests = if opt.tests || !opt.test.is_empty() {
+        // If using the --tests or --test flags, generate a list of tests and
+        // their mangled names.
+        info_at!(&opt, 3, "  Getting list of tests in {}", &package);
+        let mut tests = list_tests(&opt, &target)?;
+        if !opt.test.is_empty() {
+            tests = tests
+                .into_iter()
+                .filter(|t| opt.test.iter().any(|f| t.contains(f)))
+                .collect();
+        }
+        if tests.is_empty() {
+            Err("  No tests found")?
+        }
+        let tests: Vec<String> = tests
+            .iter()
+            .map(|t| format!("{}::{}", package, t))
+            .collect();
+
+        // then look up their mangled names in the bcfile
+        mangle_functions(&opt, &bcfile, &tests)?
+    } else if opt.backend == Backend::Seahorn {
+        // Find the entry function (mangled main)
+        let mains = mangle_functions(&opt, &bcfile, &[String::from(package) + "::main"])?;
+        match mains.as_slice() {
+            [(_, _)] => mains,
+            [] => Err("  FAILED: can't find the 'main' function")?,
+            _ => Err("  FAILED: found more than one 'main' function")?,
+        }
+    } else {
+        vec![("main".to_string(), "main".to_string())]
+    };
+    // Remove the package name from the function names (important for Klee?).
```

Comment:
The line below the comment :)
The problem starts with the name "tests" in the code, what it should really be is "functions", because in general we check functions (at the moment it can be `main` or one of the test functions).
Now, what the comment is trying to say is that after we get the mapping (`tests` maps function names to mangled names), we remove from each key the prefix `<package-name>::`, if the key has such prefix, because, I think, when we pass this name to Klee it expects it to be without the prefix.